src/common/utilities/heatmap-utils.ts

```diff
@@ -0,0 +1,39 @@
+import { Logging } from "../../containers/habit-tracker/habits/Habits";
+
+export interface HeatmapEntry {
+  month: number;
+  dayOfWeek: number;
+  logCount: number;
+  dayOfMonth: number;
+}
+
+// Function to generate heatmap data from the logging data
+export function generateHeatmapData(
+  habitLog: Logging,
+  habitName: string,
+  selectedYear: number
+): HeatmapEntry[] {
+  const habitLogData = habitLog[habitName];
+  const heatmapEntries: HeatmapEntry[] = [];
+
+  // Iterate over each day of the year
+  for (let dayOfYear = 0; dayOfYear < 365; dayOfYear++) {
+    const currentDate = new Date(selectedYear, 0, 1 + dayOfYear);
+
+    // Calculate the number of logs for the current date
+    const logCount =
+      habitLogData?.[selectedYear]?.[currentDate.toLocaleString("default", { month: "long" }).toLowerCase()]?.[currentDate.getDate()] || 0;
```

Comment:
having to do this every time where we have to get the month's name seems risky to me.
I almost want to change how we store the months in the logs from using the string to the month numbers to make data retrieval easy. We should have thought about it earlier, but it's unfortunate.